src/repository.c

```diff
@@ -1406,6 +1407,102 @@ Repository_listall_reference_objects(Repository *self, PyObject *args)
     return NULL;
 }
 
+PyObject *
+wrap_references_iterator(git_reference_iterator *iter) {
+    RefsIterator *py_refs_iter = PyObject_New(RefsIterator , &ReferenceType);
+    if (py_refs_iter)
+        py_refs_iter->iterator = iter;
+
+    return (PyObject *)py_refs_iter;
+}
+
+void
+References_iterator_dealloc(RefsIterator *iter)
+{
+    git_reference_iterator_free(iter->iterator);
+    Py_TYPE(iter)->tp_free((PyObject *)iter);
+}
+
+PyDoc_STRVAR(Repository_references_iterator_init__doc__,
+  "references_iterator_init() -> git_reference_iterator\n"
+  "\n"
+  "Creates and returns an iterator for references.");
+
+PyObject *
+Repository_references_iterator_init(Repository *self, PyObject *args)
+{
+    int err;
+    git_reference_iterator *iter;
+    RefsIterator *refs_iter;
+
+    refs_iter = PyObject_New(RefsIterator, &RefsIteratorType);
+    if (refs_iter == NULL) {
+        return NULL;
+    }
+
+    if ((err = git_reference_iterator_new(&iter, self->repo)) < 0)
+        return Error_set(err);
+
+    refs_iter->iterator = iter;
+    return (PyObject*)refs_iter;
+}
+
+static PyObject *
+wrap_and_check_reference(Repository *self, git_reference *ref) {
+    PyObject *py_ref = wrap_reference(ref, self);
+    if (py_ref == NULL) {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
```

Comment:
This is masking an error, if wrap_reference returns NULL it's because there was an error, and it must be forwarded.